### PR TITLE
Change producers approvals logic in rem.swap

### PR DESCRIPTION
**Pull request by Alladin9393**

### CODE REVIEW

File: contracts/contracts/rem.swap/src/rem.swap.cpp [hunk 1]
```diff
@@ -50,26 +50,28 @@ namespace eosio {
             s.status         = static_cast<int8_t>(swap_status::INITIALIZED);
             if (is_producer) s.provided_approvals.push_back(rampayer);
```
Comment (roman-tik) [inline contracts/contracts/rem.swap/src/rem.swap.cpp]: make it only possible for top25 bps to create records.

File: contracts/contracts/rem.swap/src/rem.swap.cpp [hunk 2]
```diff
@@ -50,26 +50,28 @@ namespace eosio {
...
 
          const vector <name> &approvals = swap_hash_it->provided_approvals;
          bool is_already_approved = std::find(approvals.begin(), approvals.end(), rampayer) == approvals.end();
 
          check(is_already_approved, "approval already exists");
 
-         swap_table.modify(*swap_hash_it, rampayer, [&](auto &s) {
-            s.provided_approvals.push_back(rampayer);
-         });
+         if (is_swap_init || is_swap_issued) {
```
Comment (roman-tik) [inline contracts/contracts/rem.swap/src/rem.swap.cpp]: remove `is_swap_issued`

File: contracts/contracts/rem.swap/src/rem.swap.cpp [hunk 3]
```diff
@@ -50,26 +50,28 @@ namespace eosio {
             s.status         = static_cast<int8_t>(swap_status::INITIALIZED);
             if (is_producer) s.provided_approvals.push_back(rampayer);
          });
-      } else if (is_producer) {
+      } else {
          check(is_producer, "block producer authorization required");
```
Comment (roman-tik) [inline contracts/contracts/rem.swap/src/rem.swap.cpp]:
```suggestion
         check(is_producer, "Top25 block producer authorization required");
```

File: contracts/contracts/rem.swap/src/rem.swap.cpp [hunk 4]
```diff
@@ -50,26 +50,28 @@ namespace eosio {
...
+         }
       }
       // moved out, because existing case when the majority of the active producers = 1
       if (is_producer) {
          cleanup_swaps();
          swap_hash_it = swap_hash_idx.find(swap_data::get_swap_hash(swap_hash));
-         bool is_status_issued = swap_hash_it->status == static_cast<int8_t>(swap_status::ISSUED);
-         if (is_swap_confirmed(swap_hash_it->provided_approvals) && !is_status_issued) {
+         bool is_status_init = swap_hash_it->status == static_cast<int8_t>(swap_status::INITIALIZED);
+         if (is_swap_confirmed(swap_hash_it->provided_approvals) && is_status_init) {
```
Comment (roman-tik) [inline contracts/contracts/rem.swap/src/rem.swap.cpp]:
```suggestion
         if (is_status_init && is_swap_confirmed(swap_hash_it->provided_approvals)) {
```

**Review by roman-tik — CHANGES_REQUESTED** [review]

File: contracts/contracts/rem.swap/src/rem.swap.cpp [hunk 5]
```diff
@@ -22,6 +22,7 @@ namespace eosio {
       swap_params_data = swap_params_table.get();
 
       check_pubkey_prefix(swap_pubkey);
+      check(is_block_producer(rampayer), "top25 block producer authorization required");
```
Comment (roman-tik) [inline contracts/contracts/rem.swap/src/rem.swap.cpp]:
```suggestion
      check(is_block_producer(rampayer), "only top25 block producers approval is recorded");
```

Comment (Alladin9393) [reply contracts/contracts/rem.swap/src/rem.swap.cpp]: @roman-tik, done.

File: contracts/contracts/rem.swap/src/rem.swap.cpp (hunk 4 above)
Comment (Alladin9393) [reply contracts/contracts/rem.swap/src/rem.swap.cpp]: @roman-tik, done.

File: contracts/contracts/rem.swap/src/rem.swap.cpp (hunk 3 above)
Comment (Alladin9393) [reply contracts/contracts/rem.swap/src/rem.swap.cpp]: @roman-tik, done.

File: contracts/contracts/rem.swap/src/rem.swap.cpp (hunk 2 above)
Comment (Alladin9393) [reply contracts/contracts/rem.swap/src/rem.swap.cpp]: @roman-tik, done.

File: contracts/contracts/rem.swap/src/rem.swap.cpp (hunk 1 above)
Comment (Alladin9393) [reply contracts/contracts/rem.swap/src/rem.swap.cpp]: @roman-tik, done.

**Comment by kushnirenko-remme** [comment]
Build_and_test

**Review by roman-tik — APPROVED** [review]

**Pull request merged**